src/libutil/hash.hh

```diff
@@ -30,6 +41,7 @@ struct Hash
     unsigned char hash[maxHashSize] = {};
 
     HashType type = htUnknown;
+    HashEncoding encoding = heUnknown;
```

Comment:
Please remove this field. The `Hash` type should not store an encoding. It creates all sorts of issues like whether `operator ==` should compare the encoding.